File: Module4/ss16_automated_testing/thuc_hanh/criteria/src/main/java/com/example/demo/model/Customer.java
package com.example.demo.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.sql.Date;

@Entity
@Table(name = "customer")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    TypeCustomer typeCustomer;

    String name;

    Date birthday;

    String id_card;

    Gender gender;

    String phone;

    String email;

    String address;
}

File: Module5/GrandMercury/GrandMercuryProject/src/app/model/customer/Customer.ts
import {TypeCustomer} from "./TypeCustomer";


export interface Customer{
  readonly id?: number;
  name?: string;
  dayOfBirth?: Date;

  typeCustomer?: TypeCustomer;

  address?: string;

  phone?: string;

  email?: string;

  avatar?: string;

  description? : string;
}
